pkg/envoy/xds_server.go

```diff
@@ -96,12 +96,11 @@ func (envoyXdsServer *XdsServer) RunManagementServer() {
 	grpcServer.GracefulStop()
 }
 
-func (envoyXdsServer *XdsServer) SetSnapshot(snapshot *cache.Snapshot, nodeID string) error {
-	err := envoyXdsServer.snapshotCache.SetSnapshot(nodeID, *snapshot)
-
-	if err != nil {
-		return err
-	}
+func (envoyXdsServer *XdsServer) GetSnapshot() (cache.Snapshot, error) {
+	//TODO: node id as a const or func arg?
```

Comment:
I think it's better as a function argument like it is now in the `SetSnapshot` function. That will allow us to keep the `envoy` package without kourier-only dependencies.